### PR TITLE
Meta: Use GitHub Actions

### LINKED ISSUE #522: Discuss new CI options
Travis-CI hasn't been welcoming for our current PRs. 

I'd like to migrate our CI work to GitHub Actions but I'm open to other suggestions.

We need to preserve our current PR builds and the build on the master branch that pushes the spec text to gh-pages.

### PR DESCRIPTION
**Pull request by leobalter**
Fixes #522

I'm still testing this out, having the open PR helps tracking the workflows.

### CODE REVIEW

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: 'ecma-262'
```
Comment (ljharb) [inline .github/workflows/preview.yml:1]: this one won't work on ecma402, deploy previews are currently hardcoded to only work on ecma262. you can remove this file for now.

File: .github/workflows/require-allow-edits.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+on: [pull_request]
+
+jobs:
+  _:
+    name: "Require “Allow Edits”"
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: ljharb/require-allow-edits@main
```
Comment (ljharb) [inline .github/workflows/require-allow-edits.yml:12]:
```suggestion
    - uses: ljharb/require-allow-edits@a200d61
```
unless you're ok with having this be a floating reference

File: .github/workflows/require-allow-edits.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ljharb) [inline .github/workflows/require-allow-edits.yml:14]:
```suggestion
```
this isn't actually required anymore, i just haven't updated all the client repos yet :-)

**Review by ljharb — COMMENTED** [review]
you may want a trailing newline on each file

File: .github/workflows/preview.yml (hunk 1 above)
Comment (leobalter) [reply .github/workflows/preview.yml]: yes, I removed this file for now. I want the preview feature but that's work for another PR.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+on: [pull_request, push]
+
+jobs:
+  build:
+    name: 'build & lint ecmarkup'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ljharb/actions/node/run@125357c
```
Comment (leobalter) [inline .github/workflows/build.yml:12]: @ljharb this is erroring, trying to install npm or something. Is there a quick link to this action code?

Comment (ljharb) [reply .github/workflows/build.yml:12]: removing the npm ci config should have resolved this.

https://github.com/ljharb/actions tho

**Comment by leobalter** [comment]
the PR checks are working, we now need to verify the deploy process, so I'm merging this.

**Pull request merged**